src/unstable/read.rs

```diff
@@ -0,0 +1,751 @@
+//! Alternate implementation of [`crate::read`].
```

Comment:
Document when to use this and when to use the main impl. Also, please remove everything we're not using internally -- we don't need a second public API.